internal/subshell/cmd/cmd.go

```diff
@@ -90,6 +90,25 @@ func (v *SubShell) WriteUserEnv(cfg sscommon.Configurable, env map[string]string
 	return nil
 }
 
+func (v *SubShell) CleanUserEnv(cfg sscommon.Configurable, envType sscommon.RcIdentification, userScope bool) error {
+	cmdEnv := NewCmdEnv(userScope)
+
+	// Clean up old entries
+	oldEnv := cfg.GetStringMap(envType.Key)
+	for k, v := range oldEnv {
+		if err := cmdEnv.unset(k, v.(string)); err != nil {
+			return err
+		}
+	}
+
+	osutils.PropagateEnv()
```

Comment:
This doesn't need an error check?

Reply:
It never had one, but I added it.  Because the Windows system function supports error handling.